src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -19,28 +20,60 @@
 
     private String mergeRemote;
     private String mergeTarget;
-    private String mergeStrategy;
+    private MergeCommand.Strategy mergeStrategy;
     private MergeCommand.GitPluginFastForwardMode fastForwardMode;
 
     /**
      * @deprecated use the new constructor that allows to set the fast forward mode.
      */
     @Deprecated
     public UserMergeOptions(String mergeRemote, String mergeTarget, String mergeStrategy) {
-        this(mergeRemote, mergeTarget, mergeStrategy, MergeCommand.GitPluginFastForwardMode.FF);
+        this(mergeTarget);
+        setMergeRemote(mergeRemote);
+        setMergeStrategy(mergeStrategy);
+        setFastForwardMode(MergeCommand.GitPluginFastForwardMode.FF);
+    }
+
+    /**
+     * @Deprecated use the new @DataBoundConstructor where you only need to supply the necessary information.
```

Comment:
Safest to keep the deprecated constructors. I do not see any reason to touch `UserMergeOptionsTest` though it seems of little value: what you really want to be testing is not whether this or that Java constructor behaves in a certain way, but whether a saved job `config.xml` with a certain configuration is loaded with the intended semantics.